src/ripple/protocol/impl/STParsedJSON.cpp

```diff
@@ -542,18 +542,20 @@ static boost::optional<detail::STVar> parseLeaf (
 
                         std::string const strValue (account.asString ());
 
-                        if (value.size () == 40) // 160-bit hex account value
-                            uAccount.SetHex (strValue);
+                        // If we have what looks like a 160-bit hex account
+                        // value, try it and see if it's valid.
+                        bool const success = (value.size () == 40)
+                            ? uAccount.SetHex (strValue)
+                            : false;
 
+                        if (!success)
```

Comment:
So the important part of this change is that we're now testing for the success of the conversion to 160-bit account value.  Previously we didn't check for success and *always* converted to RippleAddress (which I'm guessing would fail with a 160-bit hex string).  If my assessment is correct, this looks like a great change.  Nice fix.